app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,45 @@
+class PostsController < ApplicationController
+  def index
+    @user = User.includes(posts: :comments).find(params[:user_id])
+    @posts = @user.posts
+  end
```

Comment:
**OPTIONAL** - You have done well so far, but this does not fix the `n+1` issues in your project, to fix the n+1 issues, you want to make sure that all comments belonging to a particular post are fetched. I suggest using the code below:

```ruby
 @posts = @user.posts.includes(:comments)
```